controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -191,6 +191,12 @@ func (r *BareMetalHostReconciler) Reconcile(request ctrl.Request) (result ctrl.R
 				// If we are in the process of deletion, try with empty credentials
 				bmcCreds = &bmc.Credentials{}
 				bmcCredsSecret = &corev1.Secret{}
+
+				// Remove the finalizer, if present
+				if hostHasFinalizer(host) {
+					host.Finalizers = utils.FilterStringFromList(
+						host.Finalizers, metal3v1alpha1.BareMetalHostFinalizer)
```

Comment:
Using empty credentials should have allowed us to make it to `actionDelete()` where this is happening. What does the provisioner return that prevents that from happening?

Reply:
There is a bz for this bug (https://bugzilla.redhat.com/show_bug.cgi?id=1891301). When the user created the host, they specified the URL with special characters e.g. '<redfish:// ....>' and then tried to delete the host, which causes the deletion process to hang. 

Frankly, I'm not sure whether this is the correct place to remove the finalizer. Also, not sure why the malformed URL didn't get flagged.

Reply:
The log message says "failed to create provisioner", so I believe it's line 213 that is failing.